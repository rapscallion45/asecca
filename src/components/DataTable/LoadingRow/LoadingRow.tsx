import { FC } from 'react';
import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import CircularProgress from '@mui/material/CircularProgress';
import { IDataTableColumn } from '@/components/DataTable/types';

interface ILoadingRowProps {
  columns: Array<IDataTableColumn>;
  message: string;
}

/* Data Table Loading Row helper */
/* ============================= */
const LoadingRow: FC<ILoadingRowProps> = (props) => {
  const { columns, message } = props;

  return (
    <TableRow>
      <TableCell colSpan={columns.length}>
        <Box display="flex" flexDirection="column" py={7} alignItems="center">
          <CircularProgress size={30} />
          <Typography variant="body1" mt={2}>
            {message}
          </Typography>
        </Box>
      </TableCell>
    </TableRow>
  );
};

export default LoadingRow;
