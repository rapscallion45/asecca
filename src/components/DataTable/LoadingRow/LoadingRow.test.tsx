import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Table, TableBody } from '@mui/material';
import LoadingRow from './LoadingRow';
import dataTableColumns from '../../CostsConfigTable/costsConfigTableColumns';

/**
 * default test loading message
 *
 * @since 0.0.0
 */
const testMessage = 'This is a loading message';

/**
 * Loading Row Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 */
describe('Data Table Loading Row', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Table>
          <TableBody>
            <LoadingRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Message', () => {
    it('Should render the passed message text', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <LoadingRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByText(testMessage)).toBeInTheDocument();
    });
  });

  describe('Spinner', () => {
    it('Should render circular progress', async () => {
      /** Arrange */
      /** Act */
      render(
        <Table>
          <TableBody>
            <LoadingRow columns={dataTableColumns} message={testMessage} />
          </TableBody>
        </Table>
      );

      /** Assert */
      expect(screen.getByRole('progressbar')).toBeInTheDocument();
    });
  });
});
