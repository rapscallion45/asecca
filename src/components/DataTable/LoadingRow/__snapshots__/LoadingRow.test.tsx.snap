// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Loading Row Renders correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        colSpan={7}
      >
        <div
          className="MuiBox-root css-1lu6zdn"
        >
          <span
            className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
            role="progressbar"
            style={
              {
                "height": 30,
                "width": 30,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={{}}
              />
            </svg>
          </span>
          <p
            className="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
          >
            This is a loading message
          </p>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
